src/main/scala/com/cognite/sdk/scala/v1/resources/dataPoints.scala

```diff
@@ -85,6 +88,105 @@ class DataPointsResourceV1[F[_]](val requestSession: RequestSession[F])
           }
       }
 
+  def queryAggregatesById(
+      id: Long,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      granularity: String,
+      aggregateFunctions: Seq[String]
+  ): F[Map[String, Seq[DataPoint]]] =
+    requestSession
+      .sendCdf { request =>
+        request
+          .post(uri"$baseUri/list")
+          .body(
+            Items(
+              Seq(
+                QueryAggregatesById(
+                  id,
+                  inclusiveStart.toString,
+                  exclusiveEnd.toString,
+                  granularity,
+                  aggregateFunctions
+                )
+              )
+            )
+          )
+          .response(asJson[Either[CdpApiError, Items[QueryAggregatesByIdResponse]]])
+          .mapResponse {
+            case Left(value) => throw value.error
+            case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/list")
+            case Right(Right(value)) =>
+              value.items.headOption
+                .map(i => convertAggregateDataPointListToDataPointMap(i.datapoints))
+                .getOrElse(Map.empty)
+          }
+      }
+
+  def queryAggregatesByExternalId(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      granularity: String,
+      aggregateFunctions: Seq[String]
+  ): F[Map[String, Seq[DataPoint]]] =
+    requestSession
+      .sendCdf { request =>
+        request
+          .post(uri"$baseUri/list")
+          .body(
+            Items(
+              Seq(
+                QueryAggregatesByExternalId(
+                  externalId,
+                  inclusiveStart.toString,
+                  exclusiveEnd.toString,
+                  granularity,
+                  aggregateFunctions
+                )
+              )
+            )
+          )
+          .response(asJson[Either[CdpApiError, Items[QueryAggregatesByIdResponse]]])
+          .mapResponse {
+            case Left(value) => throw value.error
+            case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/list")
+            case Right(Right(value)) =>
+              value.items.headOption
+                .map(i => convertAggregateDataPointListToDataPointMap(i.datapoints))
+                .getOrElse(Map.empty)
+          }
+      }
+
+  def convertAggregateDataPointListToDataPointMap(
+      aggregateDataPoints: Seq[AggregateDataPoint]
+  ): Map[String, Seq[DataPoint]] =
+    aggregateDataPoints
+      .flatMap(adp => {
+        Seq(
+          ("average", (adp.timestamp, adp.average)),
```

Comment:
You should be able to do `.flatMap { adp => Seq("average" -> adp.average.map(DataPoint(adp.timestamp, _)), "max" -> ...) }` followed by `.groupBy(_._1)` a `Map[String, Seq[DataPoint]]`, with some filtering and getting in between to remove possible `None`s.

Ideally the flatMap would already get rid of the options, I think you could do this with `flatten`:
`.flatMap { adp => Seq(adp.average.map("average" -> DataPoint(adp.timestamp, _)), "max" -> ...).flatten }`
Then it should only be `.flatMap { adp => ... }.groupBy(_._1)`

Reply:
Thought about this some more and maybe we can just do `Map("average" -> aggregates.map(p => p.average.map(average => DataPoint(p.timestamp, average)).flatten, "max" -> ..., ...).filter { case (_, value) => value.nonEmpty }`